fMRIVolume/scripts/IntensityNormalization.sh

```diff
@@ -117,20 +140,49 @@ echo "PWD = `pwd`" >> $WD/log.txt
 
-########################################## DO WORK ########################################## 
-
-# FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
-# voxels that have data available at *ALL* time points (derived in OneStepResampling)
 FinalMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_finalmask
-${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${FinalMask}
+T1FOVMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_T1FOVmask
+${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${T1FOVMask}
+
+# Use the requested BOLD mask 
+
+if [ "${BOLDMask}" = "T1_fMRI_FOV" ] ; then
+    # FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED_fMRI_FOV" ] ; then
+    # FinalMask is a combination of once dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED2x_fMRI_FOV" ] ; then
+    # FinalMask is a combination of twice dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "fMRI_FOV" ] ; then
+    # FinalMask is the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    {FSLDIR}/bin/imcp ${InputfMRI}_mask ${FinalMask}
+
+elif
+    # No valid BOLDMask option was specified
+    log_Err_Abort "Specified BOLD mask to use (--boldmask=${BOLDMask}) is invalid!"
+fi
 
-# Create a simple summary text file of the percentage of spatial coverage of the fMRI data inside the FS-derived brain mask
+
+# Create a simple summary text file of the percentage of spatial coverage of the fMRI data inside the FS-derived brain mask and the actual mask used
 NvoxBrainMask=`fslstats ${BrainMask} -V | awk '{print $1}'`
+NvoxT1FOVMask=`fslstats ${T1FOVMask} -V | awk '{print $1}'`
 NvoxFinalMask=`fslstats ${FinalMask} -V | awk '{print $1}'`
-PctCoverage=`echo "scale=4; 100 * ${NvoxFinalMask} / ${NvoxBrainMask}" | bc -l`
-echo "PctCoverage, NvoxFinalMask, NvoxBrainMask" >| ${FinalMask}.stats.txt
-echo "${PctCoverage}, ${NvoxFinalMask}, ${NvoxBrainMask}" >> ${FinalMask}.stats.txt
+PctBrainCoverage=`echo "scale=4; 100 * ${NvoxT1FOVMask} / ${NvoxBrainMask}" | bc -l`
+PctMaskCoverage=`echo "scale=4; 100 * ${NvoxFinalMask} / ${NvoxT1FOVMask}" | bc -l`
```

Comment:
It isn't clear to me what you are trying to convey via `PctMaskCoverage`.  Since `T1FOVMask` is the most "restrictive" mask, `FinalMask` will either be equal to `T1FOVMask` or a superset of voxels that include `T1FOVMask` -- i.e., `PctMaskCoverage` >= 100%.   Is that the intent?

Reply:
Yes, indeed. It will show how larger the mask is from the T1w based one.